sentry-android-core/src/main/java/io/sentry/android/core/InternalSentrySdk.java

```diff
@@ -0,0 +1,202 @@
+      writer.name("fingerprint").value(logger, scope.getFingerprint());
+      writer.name("level").value(logger, scope.getLevel());
+      writer.name("breadcrumbs").value(logger, scope.getBreadcrumbs());
+    } catch (Exception e) {
+      options.getLogger().log(SentryLevel.ERROR, "Could not serialize scope.", e);
+      return new HashMap<>();
+    }
+
+    return data;
+  }
+
+  /**
+   * Captures the provided envelope. Compared to {@link IHub#captureEvent(SentryEvent)} this method
+   * - will not enrich events with additional data (e.g. scope) - will not execute beforeSend: it's
+   * up to the caller to take care of this - will not perform any sampling: it's up to the caller to
+   * take care of this - will enrich the envelope with a Session updates is applicable
+   *
+   * @param envelopeData the serialized envelope data
+   * @return The Id (SentryId object) of the event
+   * @throws Exception In case the provided envelope could not be parsed / is invalid
+   */
+  public static SentryId captureEnvelope(final @NotNull byte[] envelopeData) throws Exception {
+    final @NotNull IHub hub = HubAdapter.getInstance();
+    final @NotNull SentryOptions options = hub.getOptions();
+
+    final @NotNull ISerializer serializer = options.getSerializer();
+    final @Nullable SentryEnvelope envelope =
+        options.getEnvelopeReader().read(new ByteArrayInputStream(envelopeData));
+    if (envelope == null) {
+      throw new IllegalArgumentException("Envelope could not be read");
+    }
+
+    final @NotNull List<SentryEnvelopeItem> envelopeItems = new ArrayList<>();
+
+    // determine session state based on events inside envelope
+    @Nullable Session.State status = null;
+    boolean crashedOrErrored = false;
+    for (SentryEnvelopeItem item : envelope.getItems()) {
+      envelopeItems.add(item);
+
+      final SentryEvent event = item.getEvent(serializer);
+      if (event != null) {
+        if (event.isCrashed()) {
+          status = Session.State.Crashed;
+        }
+        if (event.isCrashed() || event.isErrored()) {
+          crashedOrErrored = true;
+        }
+      }
+    }
+
+    // update session and add it to envelope if necessary
+    final @Nullable Session session = updateSession(hub, options, status, crashedOrErrored);
+    if (session != null) {
+      final SentryEnvelopeItem sessionItem = SentryEnvelopeItem.fromSession(serializer, session);
+      envelopeItems.add(sessionItem);
+    }
+
+    final SentryEnvelope repackagedEnvelope =
+        new SentryEnvelope(envelope.getHeader(), envelopeItems);
+    return hub.captureEnvelope(repackagedEnvelope);
+  }
+
+  @Nullable
+  private static Session updateSession(
+      final @NotNull IHub hub,
+      final @NotNull SentryOptions options,
+      final @Nullable Session.State status,
+      final boolean crashedOrErrored) {
+    final @NotNull AtomicReference<Session> sessionRef = new AtomicReference<>();
+    hub.withScope(
```

Comment:
See https://github.com/getsentry/sentry-java/pull/2814/files#r1266374699

Reply:
Fixed here: https://github.com/getsentry/sentry-java/pull/2846

Reply:
@markushi the usage of `withScope` is still there, this should be replaced by `configureScope`